src/Microsoft.NetCore.Analyzers/Core/Security/DoNotDisableSchUseStrongCrypto.cs

```
@@ -0,0 +1,99 @@
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString s_Description = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.DoNotDisableSchUseStrongCryptoDescription),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+                DiagnosticId,
+                s_Title,
+                s_Message,
+                DiagnosticCategory.Security,
+                DiagnosticSeverity.Error,
+                isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                description: s_Description,
+                helpLinkUri: null,
+                customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(compilationStartAnalysisContext =>
+            {
+                var compilation = compilationStartAnalysisContext.Compilation;
+                var appContextTypeSymbol = compilation.GetTypeByMetadataName(WellKnownTypes.SystemAppContext);
+
+                if (appContextTypeSymbol == null)
+                {
+                    return;
+                }
+
+                var setSwitchMemberWithStringAndStringParameter = appContextTypeSymbol.GetMembers("SetSwitch").OfType<IMethodSymbol>().GetSingleOrDefaultMemberWithParameterInfos(
```

Instead of SingleOrDefault we prefer FirstOrDefault as former can throw an exception.